read-fonts/src/tables/variations.rs

```diff
@@ -498,6 +502,277 @@ impl<'a> Iterator for TupleVariationHeaderIter<'a> {
+            let scaler = tuple.compute_scalar(coords)?;
+            Some((tuple, scaler))
+        })
+    }
+
+    pub(crate) fn tuple_count(&self) -> usize {
+        self.tuple_count.count() as usize
+    }
+}
+
+/// An iterator over the [`TupleVariation`]s for a specific glyph.
+pub struct TupleVariationIter<'a, T> {
+    current: usize,
+    parent: TupleVariationData<'a, T>,
+    header_iter: TupleVariationHeaderIter<'a>,
+    serialized_data: FontData<'a>,
+    _marker: std::marker::PhantomData<T>,
+}
+
+impl<'a, T> TupleVariationIter<'a, T>
+where
+    T: TupleDelta,
+{
+    fn next_tuple(&mut self) -> Option<TupleVariation<'a, T>> {
+        if self.parent.tuple_count() == self.current {
+            return None;
+        }
+        self.current += 1;
+
+        // FIXME: is it okay to discard an error here?
+        let header = self.header_iter.next()?.ok()?;
+        let data_len = header.variation_data_size() as usize;
+        let var_data = self.serialized_data.take_up_to(data_len)?;
+
+        let (point_numbers, packed_deltas) = if header.tuple_index().private_point_numbers() {
+            PackedPointNumbers::split_off_front(var_data)
+        } else {
+            (self.parent.shared_point_numbers.clone()?, var_data)
+        };
+        Some(TupleVariation {
+            axis_count: self.parent.axis_count,
+            header,
+            shared_tuples: self.parent.shared_tuples.clone(),
+            packed_deltas: PackedDeltas::new(packed_deltas),
+            point_numbers,
+            _marker: std::marker::PhantomData,
+        })
+    }
+}
+
+impl<'a, T> Iterator for TupleVariationIter<'a, T>
+where
+    T: TupleDelta,
+{
+    type Item = TupleVariation<'a, T>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.next_tuple()
+    }
+}
+
+/// A single set of tuple variation data
+#[derive(Clone)]
+pub struct TupleVariation<'a, T> {
+    axis_count: u16,
+    header: TupleVariationHeader<'a>,
+    shared_tuples: Option<SharedTuples<'a>>,
+    packed_deltas: PackedDeltas<'a>,
+    point_numbers: PackedPointNumbers<'a>,
+    _marker: std::marker::PhantomData<T>,
+}
+
+impl<'a, T> TupleVariation<'a, T>
+where
+    T: TupleDelta,
+{
+    /// Returns true if this tuple provides deltas for all points in a glyph.
+    pub fn has_deltas_for_all_points(&self) -> bool {
+        self.point_numbers.count() == 0
+    }
+
+    pub fn point_numbers(&'a self) -> PackedPointNumbersIter<'a> {
+        self.point_numbers.iter()
+    }
+
+    /// Returns the 'peak' tuple for this variation
+    pub fn peak(&self) -> Tuple<'a> {
+        self.header
+            .tuple_index()
+            .tuple_records_index()
+            .and_then(|idx| self.shared_tuples.as_ref()?.tuples().get(idx as usize).ok())
+            .or_else(|| self.header.peak_tuple())
+            .unwrap_or_default()
+    }
+
+    // transcribed from pinot/moscato
+    /// Compute the scalar for a this tuple at a given point in design space.
```

Comment:
```suggestion
    /// Compute the scalar for this tuple at a given point in design space.
```

?

Reply:
Oops. I reworded the comment a bit and removed the "// transcribed.." part. I license my work for this repo :)